internal/controller/controllers/agent_controller.go

```diff
@@ -145,7 +147,14 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	h, err := r.Installer.GetHostByKubeKey(req.NamespacedName)
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
-			return r.deleteAgent(ctx, log, req.NamespacedName)
+			// Restoring the Host according to values from Agent
+			if err = r.restoreHostByAgent(ctx, agent); err != nil {
```

Comment:
IIRC we gated the BMH restore stuff with a flag, do we want to do the same here or should this be mostly harmless?

Reply:
So it seems the logic was originally added merely for cleanup purposes. I.e. the removal of agent without an associated host shouldn't have affected the behavior. So since a missing host shouldn't be a common scenario other than in restore flows, I'm not sure we should have a feature gate for it. I mean, it doesn't seem to have much impact on current flows. Unless you think it's safer to still have a flag to disable it?

Reply:
Okay let's leave it as is then.